(GH-701) Stop using scope.lookupvar() in templates

Phil-Friderici opened the pull request:
Add tests for existing functionality as preparation and to proof backward compatibility.

Phil-Friderici commented:
I have separated the commits to proof that the existing functionality is not changed. If prefered I can squash them into one commit.

Phil-Friderici commented:
ready for review

ghoneycutt commented:
Please squash into one commit. Thanks!

jeffmccune commented:
This looks good to me.  The tests are great!  Only question is why scope.lookup var is being left in `sensu-client.erb` and `sensu-windows-package-version.erb`?

Phil-Friderici commented:
Thanks for the positive feedback, will do the same for the other templates as well :)

Phil-Friderici commented:
Here we go:
- all appearances of scope.lookupcvar() have been removed
- all commits squashed into one

ghoneycutt commented on manifests/package.pp:
```diff
@@ -49,7 +108,7 @@
       # provider translates hyphens to dots, e.g. '0.29.0-11' maps to
       # '0.29.0.11' on the system.  This mapping is necessary to converge.
-      $pkg_version = template('sensu/sensu-windows-package-version.erb')
+      $pkg_version = regsubst($::sensu::version, '-', '.')
```
good catch!

Phil-Friderici replied:
same is true for sensu-version.erb. I have created the issue https://github.com/sensu/sensu-puppet/issues/731 to get it changed later on too.

ghoneycutt commented:
@Phil-Friderici looks like we still have a few

```
sensu-puppet$ grep -ri scope.lookupvar .
./manifests/extension.pp:  $filename = inline_template('<%= scope.lookupvar(\'source\').split(\'/\').last %>')
./manifests/handler.pp:    $filename = inline_template('<%= scope.lookupvar(\'source\').split(\'/\').last %>')
./manifests/mutator.pp:    $filename = inline_template('<%= scope.lookupvar(\'source\').split(\'/\').last %>')
./manifests/plugin.pp:      $filename = inline_template('<%= scope.lookupvar(\'name\').split(\'/\').last %>')
./manifests/plugin.pp:      $filename = inline_template('<%= scope.lookupvar(\'name\').split(\'/\').last %>')
```

While you are changing these, could you have them use .erb files. I prefer to use .erb files, even for one liners, so they can be validated.

Phil-Friderici commented:
have added the grep results to https://github.com/sensu/sensu-puppet/issues/731 as I think they should be refactored to Puppet code instead.

ghoneycutt commented:
Yes, please address that in this PR.

Solves #731

ghoneycutt commented:
@Phil-Friderici please use vagrant to confirm that all platforms still work. Here's a bash function I use to see what clients are connected, instead of poking around through the Dashboard.

```bash
function sensuclients {
  curl -s http://admin:secret@127.0.0.1:4567/clients | jq .[].name | awk -F \" '{print $2}' | awk -F \. '{print $1}' | sort
}
```

Phil-Friderici commented:
Think we should replace Ruby with Puppet code where possible. So I have replaced all occurrences of
```inline_template('<%= scope.lookupvar(\'source\').split(\'/\').last %>')``` with the much simplier to read ```basename()``` function.

Sorry, haven't used vagrant yet. Will need some time to get into the details and understand it.

Phil-Friderici commented on manifests/package.pp:
```diff
@@ -69,7 +128,7 @@
         $os_release = $facts['os']['release']['major']
         # e.g. '2012 R2' => '2012r2'
-        $pkg_url_dir = template('sensu/sensu-version.erb')
+        $pkg_url_dir = regsubst($os_release, '^(\d+)\s*[rR](\d+)', '\\1r\\2')
```
same as for line 111 above

Phil-Friderici commented:
sweet :)

ghoneycutt commented:
Released in v2.22.0
